torch/csrc/jit/tensorexpr/loopnest.cpp

```diff
@@ -1246,6 +1246,174 @@ void LoopNest::computeAt(Stmt* s, For* f) {
   temp_bufs_.emplace_back(std::make_pair(temp_buf, st->value()->dtype()));
 }
 
+class SwapReduce : public IRMutator {
+ public:
+  SwapReduce(ReduceOp* new_reduce) : new_reduce_(new_reduce) {}
+
+  Stmt* mutate(const Store* v) override {
+    if (dynamic_cast<const ReduceOp*>(v->value())) {
+      auto buf =
+          dynamic_cast<const Load*>(new_reduce_->accumulator().node())->buf();
+      auto indices =
+          dynamic_cast<const Load*>(new_reduce_->accumulator().node())
+              ->indices();
+      return new Store(buf, indices, new_reduce_, new IntImm(1));
+    }
+    return IRMutator::mutate(v);
```

Comment:
I don't think the generic mutator would do anything here: I believe we can just return `(Stmt*)v` here.

Reply:
ah I thought this would be appropriate just in case things change under the hood?

Reply:
Depends on what you want to happen here: I'd assume that SwapReducer would want to either change the statement (as per the code above) or return it completely unchanged.